src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -25,19 +25,21 @@
 class ResizeFormListener implements EventSubscriberInterface
 {
     protected $type;
-    protected $options;
+    protected $entryOptions;
+    protected $prototypeOptions;
     protected $allowAdd;
     protected $allowDelete;
 
     private \Closure|bool $deleteEmpty;
 
-    public function __construct(string $type, array $options = [], bool $allowAdd = false, bool $allowDelete = false, bool|callable $deleteEmpty = false)
+    public function __construct(string $type, array $entryOptions = [], bool $allowAdd = false, bool $allowDelete = false, bool|callable $deleteEmpty = false, array $prototypeOptions = [])
     {
         $this->type = $type;
         $this->allowAdd = $allowAdd;
         $this->allowDelete = $allowDelete;
-        $this->options = $options;
+        $this->entryOptions = $entryOptions;
         $this->deleteEmpty = \is_bool($deleteEmpty) ? $deleteEmpty : $deleteEmpty(...);
+        $this->prototypeOptions = $prototypeOptions;
```

Comment:
`$this->prototypeOptions = array_replace($options, $prototypeOptions);` (and remove the array_replace below)

Reply:
Hmmm that seems not right?

I feel like the way the CollectionType gets its prototype options is by doing the array merge on the entry options (line 30 in the CollectionType). But that isn't by definition the way the ResizeFormListener should do it as well?
I'm not sure how it was decided for the CollectionType to handle the prototype options this way. I feel like it would be just as valid to have the prototype options be completely independent of the entry options.

I think it's also important to have the way the options are used in the CollectionType and the way it's passed to the ResizeFormListener be the same. Maybe I should have prevented the double merge by storing the merged entries in a variable within the CollectionType constructor? That way if it is decided that behavior is to change, it's always the same for both the CollectionType and the ResizeFormListener.

What do you think?

Thanks so much for the feedback :)

Reply:
then we should fallback to $options when no prototype options are given, to preserve the current behavior:

`$this->prototypeOptions = $prototypeOptions ?? $options;`

and change the argument to `array $prototypeOptions = null`

Reply:
Yes totally correct. I've changed the default value for the prototypeOptions and fallback to options.
I've also changed the CollectionType constructor to prevent the double merge.

Thanks so much for your help, I'm not very good at this.